source/mod_pythia.f90

```diff
@@ -0,0 +1,398 @@
+      use, intrinsic :: iso_c_binding
+    end function pythia_defaults
+
+    subroutine pythia_setProcess(sEL,sSD,sDD,sCD,sND) bind(C, name="pythiaWrapper_setProcess")
+      use, intrinsic :: iso_c_binding
+      logical(kind=C_BOOL), value, intent(in) :: sEL, sSD, sDD, sCD, sND
+    end subroutine pythia_setProcess
+
+    subroutine pythia_setCoulomb(sCMB,tAbsMin) bind(C, name="pythiaWrapper_setCoulomb")
+      use, intrinsic :: iso_c_binding
+      logical(kind=C_BOOL), value, intent(in) :: sCMB
+      real(kind=C_DOUBLE),  value, intent(in) :: tAbsMin
+    end subroutine pythia_setCoulomb
+
+    subroutine pythia_setSeed(rndSeed) bind(C, name="pythiaWrapper_setSeed")
+      use, intrinsic :: iso_c_binding
+      integer(kind=C_INT), value, intent(in) :: rndSeed
+    end subroutine pythia_setSeed
+
+    subroutine pythia_setBeam(frameType,idA,idB,eA,eB) bind(C, name="pythiaWrapper_setBeam")
+      use, intrinsic :: iso_c_binding
+      integer(kind=C_INT), value, intent(in) :: frameType
+      integer(kind=C_INT), value, intent(in) :: idA, idB
+      real(kind=C_DOUBLE), value, intent(in) :: eA, eB
+    end subroutine pythia_setBeam
+
+    subroutine pythia_readFile(fileName) bind(C, name="pythiaWrapper_readFile")
+      use, intrinsic :: iso_c_binding
+      character(kind=C_CHAR,len=1), intent(in) :: fileName
+    end subroutine pythia_readFile
+
+    subroutine pythia_getCrossSection(sigTot,sigEl) bind(C, name="pythiaWrapper_getCrossSection")
+      use, intrinsic :: iso_c_binding
+      real(kind=C_DOUBLE), intent(inout) :: sigTot, sigEl
+    end subroutine pythia_getCrossSection
+
+    subroutine pythia_getEvent(status,code,t,theta,dEE,dPP) bind(C, name="pythiaWrapper_getEvent")
+      use, intrinsic :: iso_c_binding
+      logical(kind=C_BOOL), intent(inout) :: status
+      integer(kind=C_INT),  intent(inout) :: code
+      real(kind=C_DOUBLE),  intent(inout) :: t,theta,dEE,dPP
+    end subroutine pythia_getEvent
+
+  end interface
+
+contains
+
+subroutine pythia_parseInputLine(inLine, iLine, iErr)
+
+  use crcoall
+  use string_tools
+  use mod_settings,   only : st_debug
+  use sixtrack_input, only : sixin_echoVal
+
+  implicit none
+
+  character(len=*), intent(in)    :: inLine
+  integer,          intent(in)    :: iLine
+  logical,          intent(inout) :: iErr
+
+  character(len=:), allocatable   :: lnSplit(:)
+  integer nSplit, iBeam
+  logical spErr
+
+  ! Split the input line
+  call chr_split(inLine,lnSplit,nSplit,spErr)
+  if(spErr) then
+    write(lout,"(a)") "PYTHIA> ERROR Failed to parse input line."
+    iErr = .true.
+    return
+  end if
+
+  if(nSplit == 0) return
+
+  select case(lnSplit(1))
+
+  case("FILE")
+    if(nSplit /= 2) then
+      write(lout,"(a,i0)") "PYTHIA> ERROR Keyword FILE expected 1 argument, got ",(nSplit-1)
+      iErr = .true.
+      return
+    end if
+    pythia_settingsFile    = trim(lnSplit(2))
+    pythia_useSettingsFile = .true.
```

Comment:
Once this has been seen, it may be a good idea to forbid anyting else, and also demand it's the first line in the PYTHIA block?

Reply:
The FILE flag is not intended to replace the PYTHIA block, so no. But there needs to be some logic added to check the consistency between the file and the other settings. This needs to be added.

Reply:
Either a comment with TODO or actually fixed then.